lib/tlaw/params/type.rb

```diff
@@ -30,28 +28,26 @@ def to_doc_type
       end
 
       def convert(value)
-        validate(value) && _convert(value)
+        if (err = validation_error(value))
+          fail Nonconvertible,
+               "#{self} can't convert  #{value.inspect}: #{err}"
+        end
+        _convert(value)
       end
```

Comment:
Why this change? I am not sure that `validate && convert` is a perfectly good code (what if `validate` decides value valid, but returns `nil`?), but the core of two methods named this way seems way more natural to me than "`validation_error`, which returns some text if there is an error".

Reply:
My idea was to factorize code. The pattern `check_something or nonconvertible!(value, "not an instance of #{type}")` is there in a few places; personally I don't like it, but of course might be different for you , especially that you are the author.
Note that these are private classes and thus methods...